geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -2187,6 +2187,13 @@ boolean doClose(String reason, Throwable systemFailureCause, boolean keepAlive,
         return false;
       }
 
+      for (GatewaySender sender : allGatewaySenders) {
+        try {
+          sender.prepareForStop();
+        } catch (Exception ignore) {
+        }
```

Comment:
Is it safe to just silently ignore any exceptions seen here? Also, would this call to `sender.prepareForStop()` be better on line 2246, just before we're stopping the gateway senders? Having it up here by itself seems potentially confusing.